File: csharp/Chapter07/Project.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;

namespace com.db4o.dg2db4o.chapter7
{
    class Project
    {
        string _name;
        string _costCode;
        //IList<Employee> _employees;
        IList _employees;

         public Project(string name, string costCode)
        {
            _name = name;
            _costCode = costCode;
            //_employees = new List<Employee>();  
            _employees = new ArrayList();  

        }

        //public Project(string name, string costCode, IList<Employee> employees)
        public Project(string name, string costCode, IList employees)
        {
            _name = name;
            _costCode = costCode;
            _employees = employees;
        }

        public void assignEmployee(Employee newEmployee)
        {
            _employees.Add(newEmployee);
           if(!newEmployee.Projects.Contains(this))
               newEmployee.assignToProject(this);
        }

        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }

        public string CostCode
        {
            get
            {
                return _costCode;
            }
            set
            {
                _costCode = value;
            }
        }

        //public IList<Employee> Employees
        public IList Employees
        {
            get
            {
                return _employees;
            }
        }

        public override string ToString()
        {
            return _name + " (Project)";
        }
    }
}

File: java/src/com/db4o/dg2db4o/chapter7/Project.java

package com.db4o.dg2db4o.chapter7;

import java.util.List;
import java.util.ArrayList;

public class Project {
    
    String _name;
    String _costCode;
    List<Employee> _employees;
    
    public Project(String name, String costCode) {
        _name = name;
        _costCode = costCode;
        _employees = new ArrayList<Employee>();
    }
    
    public Project(String name, String costCode, List<Employee> employees) {
        _name = name;
        _costCode = costCode;
        _employees = employees;
    }
    
    public void assignEmployee(Employee newEmployee) {
        _employees.add(newEmployee);
        if(!newEmployee.projects().contains(this))
            newEmployee.assignToProject(this);
    }
    
    public String getName() {
        return _name;
    }
    
    public void setName(String value) {
        _name = value;
    }
    
    public String getCostCode() {
        return _costCode;
    }
    
    public void setCostCode(String value){
        _costCode = value;
    }
    
    public List<Employee> employees() {
        return _employees;
    }
    
    public String toString() {
        return _name + " (Project)";
    }
    
}
